pkg/controller/core/resourceflavor_controller.go

```diff
@@ -64,9 +87,17 @@ func (r *ResourceFlavorReconciler) Delete(e event.DeleteEvent) bool {
 	if !match {
 		return false
 	}
+	// TODO: When ResourceFlavors are used by ClusterQueues, they shouldn't been deleted directly,
+	// and notifyWatchers should be called after deletion.
+	defer r.notifyWatchers(flv)
+
 	log := r.log.WithValues("resourceFlavor", klog.KObj(flv))
 	log.V(2).Info("ResourceFlavor delete event")
 	r.cache.DeleteResourceFlavor(flv)
+
```

Comment:
if `DeleteResourceFlavor` returns a bool to indicate if at least one CQ changed its status from pending to ready, if true we call `r.qManager.Broadcast()`